tests/queries/0_stateless/02981_nested_bad_types.sql

```diff
@@ -0,0 +1,54 @@
+
+
+
+select ['42']::Array(FixedString(1000000)); -- {serverError ILLEGAL_COLUMN}
+select ['42']::Array(FixedString(1000000)); -- {serverError ILLEGAL_COLUMN}
+select [[['42']]]::Array(Array(Array(FixedString(1000000)))); -- {serverError ILLEGAL_COLUMN}
+select map('a', '42')::Map(String, FixedString(1000000)); -- {serverError ILLEGAL_COLUMN}
+select map('a', map('b', ['42']))::Map(String, Map(String, Array(FixedString(1000000)))); -- {serverError ILLEGAL_COLUMN}
+select tuple('a', '42')::Tuple(String, FixedString(1000000)); -- {serverError ILLEGAL_COLUMN}
+select tuple('a', [map('b', '42')])::Tuple(String, Array(Map(String, FixedString(1000000)))); -- {serverError ILLEGAL_COLUMN}
+select '42'::Variant(UInt64, FixedString(1000000)) settings allow_experimental_variant_type=1; -- {serverError ILLEGAL_COLUMN}
+
+
+create table test (x Array(FixedString(1000000))) engine=Memory; -- {serverError ILLEGAL_COLUMN}
+create table test (x Array(Array(FixedString(1000000)))) engine=Memory; -- {serverError ILLEGAL_COLUMN}
+create table test (x Map(String, FixedString(1000000))) engine=Memory; -- {serverError ILLEGAL_COLUMN}
+create table test (x Map(String, Map(String, FixedString(1000000)))) engine=Memory; -- {serverError ILLEGAL_COLUMN}
+create table test (x Tuple(String, FixedString(1000000))) engine=Memory; -- {serverError ILLEGAL_COLUMN}
+create table test (x Tuple(String, Array(Map(String, FixedString(1000000))))) engine=Memory; -- {serverError ILLEGAL_COLUMN}
+
+
+select [42]::Array(Variant(String, UInt64)); -- {serverError ILLEGAL_COLUMN}
+select [42]::Array(Variant(String, UInt64)); -- {serverError ILLEGAL_COLUMN}
+select [[[42]]]::Array(Array(Array(Variant(String, UInt64)))); -- {serverError ILLEGAL_COLUMN}
+select map('a', 42)::Map(String, Variant(String, UInt64)); -- {serverError ILLEGAL_COLUMN}
+select map('a', map('b', [42]))::Map(String, Map(String, Array(Variant(String, UInt64)))); -- {serverError ILLEGAL_COLUMN}
+select tuple('a', 42)::Tuple(String, Variant(String, UInt64)); -- {serverError ILLEGAL_COLUMN}
+select tuple('a', [map('b', 42)])::Tuple(String, Array(Map(String, Variant(String, UInt64)))); -- {serverError ILLEGAL_COLUMN}
+
+create table test (x Array(Variant(String, UInt64))) engine=Memory; -- {serverError ILLEGAL_COLUMN}
+create table test (x Array(Array(Variant(String, UInt64)))) engine=Memory; -- {serverError ILLEGAL_COLUMN}
+create table test (x Map(String, Variant(String, UInt64))) engine=Memory; -- {serverError ILLEGAL_COLUMN}
+create table test (x Map(String, Map(String, Variant(String, UInt64)))) engine=Memory; -- {serverError ILLEGAL_COLUMN}
+create table test (x Tuple(String, Variant(String, UInt64))) engine=Memory; -- {serverError ILLEGAL_COLUMN}
+create table test (x Tuple(String, Array(Map(String, Variant(String, UInt64))))) engine=Memory; -- {serverError ILLEGAL_COLUMN}
+
```

Comment:
```suggestion
set allow_experimental_variant_type=1;

select '42'::Variant(UInt64, FixedString(1000000)) settings allow_experimental_variant_type=1; -- {serverError ILLEGAL_COLUMN}
create table test (x Variant(String, FixedString(1000000))) engine=Memory; -- {serverError ILLEGAL_COLUMN}
create table test (x Variant(String, Map(String, FixedString(1000000)))) engine=Memory; -- {serverError ILLEGAL_COLUMN}
```

This way we can be sure that the query fails because of the nested types in variant and not because of variant itself.

Reply:
There is a check above allong with other FixedString checks:
```
select '42'::Variant(UInt64, FixedString(1000000)) settings allow_experimental_variant_type=1; -- {serverError ILLEGAL_COLUMN}
```
But yes, let's add more checks like you suggested